### ISSUE TITLE
If geom in first row returned by a sql statement with a function on the geom is NULL, all geoms become NULL

### ISSUE THREAD

**Issue by theroggy** [geofileops/geofileops#308]
Eg. running a negative buffer on a file where the geometry for the first geometry that results in NULL will result in getting NULL for all geometries in the output file.

In the script below it is illustrated that
  - in the first sql_stmt, the geom in the first row becomes NULL and this results in all rows getting a NULL geometry
  - in the second one, the order by makes sure the first row is not NULL, which results in a correct output.

The issue boils down to gdal.

```
from pathlib import Path

import geofileops as gfo
import geopandas as gpd
from shapely import Polygon

# Prepare test data
data = [
    {
        "descr": "polygon1",
        "geometry": Polygon([(0, 0), (0, 5), (5, 5), (5, 0), (0, 0)]),
    },
    {
        "descr": "polygon2",
        "geometry": Polygon([(0, 0), (0, 15), (15, 15), (15, 0), (0, 0)]),
    },
]
input_gdf = gpd.GeoDataFrame(data=data, crs=31370)  # type: ignore

tmp_path = Path("C:/temp")
input_path = tmp_path / f"test.gpkg"
gfo.to_file(input_gdf, input_path)
layer = gfo.get_only_layer(input_path)
distance = -5
sql_stmt = f"""
    SELECT ST_CollectionExtract(
               ST_Buffer({{geometrycolumn}}, {distance}, 5), 3
           ) AS geom
          {{columns_to_select_str}}
      FROM "{layer}" layer
"""

# If first row becomes NULL due to the function, all geometries become NULL.
result_path = tmp_path / f"test_select_null.gpkg"
gfo.select(
    input_path=input_path,
    output_path=result_path,
    sql_stmt=sql_stmt,
)
result_gdf = gfo.read_file(result_path)
print(result_gdf)

# When the rows are ordered so the first row is not NULL, result is OK.
result_path = tmp_path / f"test_select_null_ordered.gpkg"
sql_stmt = f"SELECT * FROM ({sql_stmt}) ORDER BY geom IS NULL"
gfo.select(
    input_path=input_path,
    output_path=result_path,
    sql_stmt=sql_stmt,
)
result_gdf = gfo.read_file(result_path)
print(result_gdf)
```

**Comment by theroggy** [comment]
Issue opened in gdal: [#308](https://github.com/OSGeo/gdal/issues/8587)

**Comment by theroggy** [comment]
Fixed in gdal >= 3.8
